src/app/charts/line-graph.component.ts

```diff
@@ -182,4 +178,69 @@ export class LineGraphComponent {
     return [slope, intercept, rSquare];
   }
 
+  private drawMinMax(): void {
+    if (self.min || self.max) {
+      // Prepare standard variables
+      var x1 = self.xRange[1];
+      var x2 = self.xRange[0];
+      var y = D3.max(self.yData)/2;
+      var xscale = D3.scaleBand()
+                    .range([0, self.width])
+                    .padding(0)
+                    .domain(self.extractedData.map(function(d) { return d.date; }));
+
+      if (self.min && self.minVal) {
+        // Draw min threshold line
+        var minData = [{"date":x1, "value": self.minVal }, {"date": x2, "value": self.minVal }];
+        self.drawLine(minData, "minline");
+
+        // Prepare data for bar graph
+        var minBars = self.extractedData.map(function(datum) {
+            var val;
```

Comment:
Stylistically this is a bit convoluted. This would be much more concise for these two lines:
```
let val = this.minVal > datum["value"] ? y * 2 : 0;
```